src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NTS.Domain.Setup.Entities;
+using NTS.Persistence.Setup;
+
+namespace NTS.Persistence.Adapters;
+
+public class SetupLapRepository : SetRepository<Lap, SetupState>
+{
+    public SetupLapRepository(IStore<SetupState> store) : base(store)
+    {
+        state = store.Load().Result;
```

Comment:
Perhaps I didn't emphasize enough that we don't want `Task.Result` in the codebase. At best it negates all benefits from asynchronous code, at worst it can halt the app.

Furthermore, constructors **should not** contain asynchronous logic. It's especially dangerous to call `.Result` here, because this is a **Dependency**, i.e. a service managed by the Dependency Injection Container. The container uses this constructor in ways you do not undestand and very often they are used during app startup, when configurations have not been executed yet. If this is a call to an actual database it will probably throw connectivity-related exceptions. Constructors are meant for a lightweight object initialization that can be used by the C# runtime on demand. **It's not a coincidence that they do not support async syntax**